server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -84,15 +92,59 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		return nil, err
 	}
 
-	latestDeployment = requestedDeployment.BuildPersistableInfo()
-	if persistErr := p.persistLatestDeployment(ctx, latestDeployment); persistErr != nil {
-		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
+	// log error and continue deploys if any of the post deploy task fails
+	if err := p.runPostDeployTasks(ctx, requestedDeployment, scope); err != nil {
+		logger.Error(ctx, "error running post deploy task", key.ErrKey, err)
 	}
 
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
-	return latestDeployment, err
+	return requestedDeployment.BuildPersistableInfo(), err
+}
+
+func (p *Deployer) runPostDeployTasks(ctx workflow.Context, deployment terraform.DeploymentInfo, scope metrics.Scope) error {
```

Comment:
nit: this file is like 300 lines of code and the tests are just going to be harder to maintain.  I'd like to see this moved, but it's not blocking.